libs/src/api/user.js

```diff
@@ -584,7 +584,12 @@ class Users extends Base {
   async _waitForCreatorNodeEndpointIndexing (userId, creatorNodeEndpoint) {
     let isUpdated = false
     while (!isUpdated) {
-      const user = (await this.discoveryProvider.getUsers(1, 0, [userId]))[0]
+      const userList = (await this.discoveryProvider.getUsers(1, 0, [userId]))
+      if (!userList || userList.length === 0) {
```

Comment:
what if that user never gets indexed by dicovery node? we should probably add a max timeout here too

Reply:
i agree but i want to cc @raymondjacobson - was there intentionally no timeout here? As in if this particular update is never processed, should the user legitimately be stuck forever?

Reply:
i think it adds an assumption we weren't making before by adding a timeout. maybe worth discussion, but not here. i think this PR is a straight forward bugfix--

I would rewrite this though as

```
let isUpdated = false
while (!isUpdated) {
  const userList = (await this.discoveryProvider.getUsers(1, 0, [userId]))
  if (userList) {
    const user = userList[0]
    if (user && user.creator_node_endpoint === creatorNodeEndpoint) isUpdated = true
  }
  await Utils.wait(500)
}

```

Reply:
updated w/this pattern